src/geometry/ArborX_DetailsAlgorithms.hpp

```diff
@@ -112,6 +110,53 @@ KOKKOS_INLINE_FUNCTION decltype(auto) returnCentroid(Geometry const &geometry)
                             Geometry>::apply(geometry);
 }
 
+template <class Geometry, class Space = Kokkos::HostSpace>
+struct GeometryReducer
```

Comment:
I think `ArborX_DetailsAlgorithms.hpp` is the right place for it. When we do the `src/geometry` refactoring, it would be in a separate file.